ui/src/app/machines/components/MachineHeaderForms/ActionFormWrapper/TagForm/AddTagForm/AddTagForm.tsx

```diff
@@ -0,0 +1,115 @@
+import { useEffect, useState } from "react";
+
+import { Col, Row } from "@canonical/react-components";
+import { useDispatch, useSelector } from "react-redux";
+import * as Yup from "yup";
+
+import FormikField from "app/base/components/FormikField";
+import FormikForm from "app/base/components/FormikForm";
+import type { Machine } from "app/store/machine/types";
+import type { RootState } from "app/store/root/types";
+import { actions as tagActions } from "app/store/tag";
+import tagSelectors from "app/store/tag/selectors";
+import type { CreateParams, Tag } from "app/store/tag/types";
+import { NodeStatus } from "app/store/types/node";
+import KernelOptionsField from "app/tags/components/KernelOptionsField";
+
+type Props = {
+  machines: Machine[];
+  name: string | null;
+  onClose: (tag: Tag | null) => void;
```

Comment:
maybe `onTagCreated` would be a bit more descriptive? `onClose` we usually use for the cancel button, but its use is a bit different here

Reply:
Good point! Done.